File: 2559-count-vowel-strings-in-ranges/2559-count-vowel-strings-in-ranges.cpp
class Solution {
public:
    vector<int> vowelStrings(vector<string>& words, vector<vector<int>>& queries) {
        int n = words.size();
        vector<int> prefix(n, 0), ans;

        string vow = "aeiou";
        if(vow.find(words[0][0]) != string::npos && vow.find(words[0].back()) != string::npos) {
            prefix[0] = 1;
        }

        for(int i = 1; i < n; i++) {
            if(vow.find(words[i][0]) != string::npos && vow.find(words[0].back()) != string::npos) {
                prefix[i] = prefix[i - 1] + 1;
            }
            else {
                prefix[i] = prefix[i - 1];
            }
        }

        for(vector<int> &q: queries) {
            int l = q[0];
            int r = q[1];

            int temp = (l != 0) ? prefix[r] - prefix[l - 1] : prefix[r];
            ans.push_back(temp);
        }

        return ans;
    }
};

File: 2559-count-vowel-strings-in-ranges/2559-count-vowel-strings-in-ranges.js
/**
 * @param {string[]} words
 * @param {number[][]} queries
 * @return {number[]}
 */

const isVowel = (ch) => 'aeiou'.includes(ch);

var vowelStrings = function(words, queries) {
   let prefix = new Array(words.length + 1).fill(0);

   for(let i = 0; i < words.length; i++) {
        let curr = words[i];
        if(isVowel(curr[0]) && isVowel(curr[curr.length - 1])) {
            prefix[i + 1] += prefix[i] + 1;
        }
        else prefix[i + 1] = prefix[i];
   } 

   const ans = []
   for(const [l, r] of queries) {
        ans.push(prefix[r + 1] - prefix[l])
   }

   return ans;
};